controller/src/beerocks/master/tasks/dynamic_channel_selection_task.h

```diff
@@ -90,6 +91,8 @@ class dynamic_channel_selection_task : public task, public std::enable_shared_fr
 
     bool m_is_single_scan_pending = false;
     bool m_is_single_scan         = false;
+
+    static int m_scanning;
```

Comment:
```suggestion
    static int m_scanning_task_id = -1; // Currently scanning task id. -1 means no scan in progress.
```

Reply:
static initialization is done inside the cpp file.
I added a define for clarity